### DISCUSSION TITLE
Rework the concept of lenient/strict mode

### DISCUSSION THREAD

**Discussion by oliverklee** [MyIntervals/PHP-CSS-Parser#459] [General]
Continuing from https://github.com/MyIntervals/PHP-CSS-Parser/discussions/454#discussioncomment-8430057:

> I think lenient/strict mode is a misnomer. Lenient mode is what browsers do: silently drop invalid constructs and move on to the next (though info is made available in the browser developer tools). Strict mode effectively means throw an exception upon encountering invalid syntax: so it is more like a debugging aid (to help find why some elements are not being parsed). On that basis, I think the switch should be retained, but renamed 'debug' to make its intent clear.

So, to help debugging, maybe we can have a single defined behavior (without any switches) and also provide a log of any problems. Would that solve both problems (reduce complexity and the required maintenance/testing work while also allowing users to debug problems with their CSS)?

What do you think, @sabberworm @JakeQZ ?

**Comment by JakeQZ** [comment]
I think there would still need to be a switch to enable logging (disabled by default), since there are different use cases: the developer wants to know about errors in their own CSS; in a different type of project, the deveoper does not want to know about CSS errors in user input, if, say, they are providing an online tool.

But otherwise yes, I think so.  We would not need to test creation of log file messages for specific CSS errors, just that the enable/disable switch works at a basic level.

**Comment by oliverklee** [comment]
> I think there would still need to be a switch to enable logging (disabled by default)

Yes, absolutely makes sense.

**Comment by JakeQZ** [comment]
> > I think there would still need to be a switch to enable logging (disabled by default)
> 
> Yes, absolutely makes sense.

But see https://github.com/MyIntervals/PHP-CSS-Parser/issues/461#issuecomment-1942925155

**Comment by JakeQZ** [comment]
xf #461 #464

**Reply by JakeQZ** [reply]
Thought this would cross-link these issues, but nothing's appeared.
